netlink-request/src/lib.rs

```diff
@@ -143,8 +143,8 @@ mod linux {
                     .map_err(|e| io::Error::new(io::ErrorKind::InvalidData, e))?;
                 match response.payload {
                     // We've parsed all parts of the response and can leave the loop.
-                    NetlinkPayload::Ack(_) | NetlinkPayload::Done => return Ok(responses),
-                    NetlinkPayload::Error(e) => return Err(e.into()),
+                    NetlinkPayload::Done(_) => return Ok(responses),
+                    NetlinkPayload::Error(e) if e.code.is_some() => return Err(e.into()),
                     _ => {},
                 }
```

Comment:
These are the two breaking changes in `netlink-packet-core`:
- https://github.com/rust-netlink/netlink-packet-core/commit/0c75fb5
- https://github.com/rust-netlink/netlink-packet-core/commit/52732b3

I'm not sure about the latter: previously we've interpreted `Ack` as "return stored responses". Ack is now a variant of error, so it feels weird to 100% copy the logic.

@mcginty do know if we're guaranteed to get `Done` even if `Ack` (error with `None` error code in new version) is delivered?